sql: TestTemporaryObjectCleaner failed

sql.TestTemporaryObjectCleaner [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21336300?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21336300?buildTab=artifacts#/) on master @ [caaa4e6b8563c8016ae2c47a68602a4f26e91606](https://github.com/cockroachdb/cockroach/commits/caaa4e6b8563c8016ae2c47a68602a4f26e91606):


```
=== RUN   TestTemporaryObjectCleaner
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestTemporaryObjectCleaner2526376275
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:637: DRPC is enabled (via testserver factory defaults)
    temporary_schema_test.go:196: condition failed to evaluate within 45s: from temporary_schema_test.go:212: expected 1 temp schemas, found 0
        
        goroutine dump: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/goroutine_dump_1849987045.txt
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/68358425a2d75ead146501979d1133be/logTestTemporaryObjectCleaner2526376275
--- FAIL: TestTemporaryObjectCleaner (58.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTemporaryObjectCleaner.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63602

--- a/pkg/sql/temporary_schema_test.go
+++ b/pkg/sql/temporary_schema_test.go
@@ -155,6 +155,17 @@ func TestTemporaryObjectCleaner(t *testing.T) {
 				UseDatabase: "defaultdb",
 				Knobs:       knobs,
 				Settings:    settings,
+				// With an auto-injected virtual cluster, both the system tenant
+				// and the application layer run a TemporaryObjectCleaner per
+				// pod. They share the TempObjectsCleanupCh testing knob, and
+				// the application-layer cleaners run concurrently with no
+				// coordination — exposing a race where a cleaner with a stale
+				// SQL instance reader cache deletes the still-active session's
+				// temp schema. See #169912 for the underlying production race;
+				// disable VC injection here so this test exercises only the
+				// meta1-leaseholder-gated system tenant path it was written
+				// for.
+				DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(169912),
 			},
 		},
 	)

